input/input_test.go

```diff
@@ -0,0 +1,376 @@
+								Username:    "testUSer",
+								AccessToken: "1234adsr",
+							},
+						},
+					},
+				},
+				Targets: []Target{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "incorrect input with some missing fields",
+			c: &Config{
+				Loglevel: "info",
+				Logpath:  "./test.yaml",
+				AuditJobs: []AuditJob{
+					{
+						Name:            "",
+						PollingInterval: "30s",
+						Output: Output{
+							TargetName: []string{"testtarget1"},
+						},
+						RepositoryType: "github",
+						RepositoryName: "",
+						RepositoryConfig: RepositoryConfig{
+							RepositoryURL: "https://testurl",
+							RepositoryCredentials: RepositoryCredentials{
+								Username:    "testUSer",
+								AccessToken: "",
+							},
+						},
+					},
+				},
+				Targets: []Target{
+					{
+						Name: "testtarget1",
+						Type: "elasticsearch",
+						TargetConfig: TargetConfig{
+							"host":     "test",
+							"protocol": "http",
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.c.validate(); (err != nil) != tt.wantErr {
+				t.Errorf("Config.validate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_checkPollingIntervalFormat(t *testing.T) {
+	type args struct {
+		pi string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "checking interval ending with s seconds correctly",
+			args:    args{pi: "10s"},
+			wantErr: false,
+		},
+		{
+			name:    "checking interval ending with m minutes correctly",
+			args:    args{pi: "10m"},
+			wantErr: false,
+		},
+		{
+			name:    "checking interval ending with h hours correctly",
+			args:    args{pi: "10h"},
+			wantErr: false,
+		},
+		{
+			name:    "checking interval ending with d days correctly",
+			args:    args{pi: "10d"},
+			wantErr: false,
+		},
+		{
+			name:    "checking interval ending with other than s,m,d,h",
+			args:    args{pi: "10a"},
+			wantErr: true,
+		},
+		{
+			name:    "checking interval starting with non numeric",
+			args:    args{pi: "a10a"},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := checkPollingIntervalFormat(tt.args.pi); (err != nil) != tt.wantErr {
+				t.Errorf("checkPollingIntervalFormat() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_isNumeric(t *testing.T) {
+	type args struct {
+		s string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "correct numeric string",
+			args: args{s: "10"},
+			want: true,
+		},
+		{
+			name: "incorrect numeric string",
+			args: args{s: "1012as"},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := isNumeric(tt.args.s); got != tt.want {
+				t.Errorf("isNumeric() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestInitConfig(t *testing.T) {
```

Comment:
should the name be Test_InitConfig to be consistent?

Reply:
Hi @aswinbh  , golang unit test prefers this way, Test + functionName.